torch/_library/custom_ops.py

```diff
@@ -640,12 +668,17 @@ def adinplaceorview_impl(keyset, *args, **kwargs):
                         keyset & _C._after_ADInplaceOrView_keyset, *args, **kwargs
                     )
 
-            lib.impl(
-                self._name,
-                adinplaceorview_impl,
-                "ADInplaceOrView",
-                with_keyset=True,
-            )
+            if need_python_dispatch:
+                self._opoverload.py_impl(_C.DispatchKey.ADInplaceOrView)(
+                    adinplaceorview_impl
+                )
+            else:
+                lib.impl(
+                    self._name,
+                    adinplaceorview_impl,
+                    "ADInplaceOrView",
+                    with_keyset=True,
+                )
```

Comment:
I feel like we should align the API of py_impl and lib.impl. Otherwise it's kind of annoying

Reply:
Doesn't need to happen in this PR, just a suggestion